components/proxy/src/main/kotlin/com/hotels/styx/services/OriginsConfigConverter.kt

```diff
@@ -98,17 +98,22 @@ internal class OriginsConfigConverter(
                 Pair(serviceName, healthCheckService(serviceName, appId, healthCheckConfig))
             }
 
-    internal fun healthCheckService(serviceName: String, appId: String, healthCheckConfig: HealthCheckConfig): ProviderObjectRecord {
-        assert(healthCheckConfig.isEnabled)
-        assert(healthCheckConfig.uri().isPresent)
-
-        val str = MAPPER.writeValueAsString(HealthCheckConfiguration(
-                appId,
-                healthCheckConfig.uri().get(),
-                healthCheckConfig.timeoutMillis(),
-                healthCheckConfig.intervalMillis(),
-                healthCheckConfig.healthyThreshold(),
-                healthCheckConfig.unhealthyThreshold()))
+    internal fun healthCheckService(serviceName: String, appId: String, healthCheckConfig: HealthCheckConfig?): ProviderObjectRecord {
```

Comment:
is there a good reason to change this signature? Why is it okay to take in a nullable healthCheckConfig here now?

Reply:
better to use this method instead of the currently defined one? 
```
    internal fun healthCheckServices(apps: List<BackendService>) = apps
            .filter(::isHealthCheckConfigured)
            .mapNotNull {
                it.healthCheckConfig()?.let { hc ->
                    it.id() to hc
                }
            }
            .map { (id, healthCheckConfig) ->
                val appId = id.toString()
                val serviceName = "$appId-monitor"
                
                Pair(serviceName, healthCheckService(serviceName, appId, healthCheckConfig))
            }
```